scrapi/database.py

```diff
@@ -14,20 +15,67 @@
 logger = logging.getLogger(__name__)
 
 
-try:
-    connection.setup(settings.CASSANDRA_URI, settings.CASSANDRA_KEYSPACE)
-    management.create_keyspace(settings.CASSANDRA_KEYSPACE, replication_factor=1, strategy_class='SimpleStrategy')
-except NoHostAvailable:
-    logger.error('Could not connect to Cassandra, expect errors.')
-    if settings.RECORD_HTTP_TRANSACTIONS or 'cassandra' in settings.NORMALIZED_PROCESSING or 'cassandra' in settings.RAW_PROCESSING:
-        raise
+class DatabaseManager(object):
+    def __init__(self, uri=None, keyspace=None):
+        self._models = []
+        self._setup = False
 
+        self.uri = uri or settings.CASSANDRA_URI
+        self.keyspace = keyspace or settings.CASSANDRA_KEYSPACE
 
-def cassandra_init(*args, **kwargs):
-    if connection.cluster is not None:
-        connection.cluster.shutdown()
-    if connection.session is not None:
-        connection.session.shutdown()
-    connection.setup(settings.CASSANDRA_URI, settings.CASSANDRA_KEYSPACE)
+    def setup(self, force=False, throw=False):
+        if self._setup and not force:
+            return True
 
-worker_process_init.connect(cassandra_init)
+        try:
+            connection.setup(self.uri, self.keyspace)
+            management.create_keyspace(self.keyspace, replication_factor=1, strategy_class='SimpleStrategy')
+            for model in self._models:
+                model.__keyspace__ = self.keyspace
+                management.sync_table(model)
+        except NoHostAvailable:
+            logger.error('Could not connect to Cassandra, expect errors.')
+            return False
+
+        # Note: return values are for test skipping
+        self._setup = True
+        return True
+
+    def tear_down(self):
+        if not self._setup:
+            logger.warning('Attempting to tear down a database that was never setup')
+
+        if connection.cluster is not None:
+            connection.cluster.shutdown()
+        if connection.session is not None:
+            connection.session.shutdown()
+
+        self._setup = False
+
+    def clear_keyspace(self, force=False):
```

Comment:
can you make this fail if the keyspace is settings.CASSANDRA_KEYSPACE? We really don't want that to be deletable.